### PR TITLE
feat: pie story with full data flow

### PR DESCRIPTION
**Pull request by simchaNielsen**

### CODE REVIEW

File: plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export type ChartProps<R extends string = string, C extends string = string, M e
   queriesData: {
     data: QueryData<R, C, M>[];
   }[];
-  queryData: {
+  queryData?: {
```
Comment (MatanBobi) [inline plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts:49]: How come this is nullable?

File: stories/plugin-chart-pie/PieChart.stories.tsx [hunk 2]
```diff
@@ -49,31 +50,26 @@ export default {
 const DefaultTemplate = args => (
   <ThemeProvider theme={supersetTheme}>
     <PieChart
-      {...args}
-      data={transformProps(({ ...legendTopPercentage, queriesData: args.queriesData } as unknown) as ChartProps).data}
+      {...transformProps({
```
Comment (MatanBobi) [inline stories/plugin-chart-pie/PieChart.stories.tsx:53]: I'd pull it to a function.. there are too many spreads and object creations here IMO..

File: stories/utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export const extractArgs = <A extends object>(args: A, other: A): A =>
+  Object.entries(args).reduce(
+    (acc, [key, val]) => {
```
Comment (MatanBobi) [inline stories/utils.ts:3]: Are there better names you can put here? something that will explain what's happening?

File: stories/utils.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export const extractArgs = <A extends object>(args: A, other: A): A =>
```
Comment (MatanBobi) [inline stories/utils.ts:1]: `other` seems like a reduced name, can you please call it `extraArgs` or something?

**Review by MatanBobi — APPROVED** [review]

File: plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts (hunk 1 above)
Comment (simchaNielsen) [reply plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts:49]: in new superset version this property not in usage anymore

**Pull request merged**